src/lib/decorate-text.jsx

```diff
@@ -0,0 +1,30 @@
+const React = require('react'); // eslint-disable-line
+const reactStringReplace = require('react-string-replace');
+
+/**
+ * Helper method that replaces @mentions and #hashtags in plain text
+ * 
+ * @param  {string}   text     string to convert
+ * @return {string}            string with links for @mentions and #hashtags
+ */
+module.exports = text => {
+    let replacedText;
+    
+    // Match @-mentions
+    replacedText = reactStringReplace(text, /@(\w+)/g, (match, i) => (
```

Comment:
I don't think `\w` includes hyphens or underscores. The username regex we use to validate usernames during registration is `/^[\w_-]+/`.